blog/2023-06-22-languages-perspective.md

```diff
@@ -0,0 +1,81 @@
+
+## Metaprogramming capabilities
+
+At the most basic level, the Wing compiler can synthesize any file or directory structure. When used for creating cloud applications, these files are typically a set of infrastructure definitions (such as CloudFormation, Terraform or Kubernetes manifests), Dockerfiles, function code bundles, deployment workflows, and any other artifacts that are needed in order to deliver this application to the cloud.
+
+## Immutability by default
+
+Unexpected mutations are a common source of programmer error in large-scale systems, especially when sharing data across abstraction boundaries. A language-level guarantee that state cannot change offers opportunities for caching and runtime optimizations. To that end, each of Wing’s collection types (`Array`, `Set`, `Map`) has a mutable variant (`MutArray`, `MutSet`, `MutMap`). In addition, variables cannot be reassigned to unless it is opted into with the `var` keyword.
+
+## No null pointers
+
+The initialization state of every variable in Wing can be determined by its type, and every variable must be initialized before it can be used. A common use of null in other languages, to represent an empty state or sentinel value, is subsumed by the language’s more general-purpose `Optional` type, and a selection of syntaxes for easily manipulating `Optional`-typed values.
+
+## Interop with TypeScript/JavaScript
+
+To avoid building an entire language ecosystem from scratch, Wing assumes a JavaScript-based runtime environment, both for dynamically generating configuration files and as the default way to run inflight application code. This makes it straightforward to reuse code from [npm]'s ecosystem of 3M+ packages, and to “escape hatch” into the underlying JavaScript runtime environment for cases where Wing’s abstractions are not sufficient.
+
+[npm]: https://www.npmjs.com/
+
+## JSON as a first-class type
+
+JSON is one of the most widely used protocols for sending and receiving information between internet services due to its simplicity and human-readability. As the lingua franca of the cloud, Wing includes it as a built-in data type, with `Json` and `MutJson` variants. Some of the features enabled by the JSON type to make life easier for developers are schema validation and the ability to parse JSON values into well-typed structs with compiler support.*
+
+## Local type inference
+
+To save some quantity of programmer key-pressing, Wing supports local type inference: signatures of functions and classes always require type annotation, but within the body of a function, many variables can be declared without any annotations, and Wing will infer the variable’s type from its initialization or its usage.
```

Comment:
Is it indeed "initialization or its usage" or just "initialization"? Just couldn't quickly think of an example